src/libraries/System.Private.Xml/src/System/Xml/Schema/XsdDateTime.cs

```diff
@@ -638,62 +628,42 @@ private void PrintTime(StringBuilder sb)
                     fractionDigits--;
                     fraction /= 10;
                 }
-                text = new char[fractionDigits + 1];
-                text[0] = '.';
-                IntToCharArray(text, 1, fraction, fractionDigits);
-                sb.Append(text);
+
+                vsb.Append('.');
+                vsb.AppendSpanFormattable(fraction, format: "D" + fractionDigits, provider: null);
             }
         }
 
         // Serialize time zone
-        private void PrintZone(StringBuilder sb)
+        private void PrintZone(ref ValueStringBuilder vsb)
         {
-            char[] text;
             switch (InternalKind)
             {
                 case XsdDateTimeKind.Zulu:
-                    sb.Append('Z');
+                    vsb.Append('Z');
                     break;
                 case XsdDateTimeKind.LocalWestOfZulu:
-                    text = new char[s_lz_zz_zz];
-                    text[0] = '-';
-                    ShortToCharArray(text, s_Lz_, ZoneHour);
-                    text[s_lz_zz] = ':';
-                    ShortToCharArray(text, s_lz_zz_, ZoneMinute);
-                    sb.Append(text);
-                    break;
+                    {
+                        vsb.Append('-');
+                        vsb.AppendSpanFormattable(ZoneHour, format: "D2", provider: null);
+                        vsb.Append(':');
+                        vsb.AppendSpanFormattable(ZoneMinute, format: "D2", provider: null);
+                        break;
+                    }
```

Comment:
Nit: these braces and this extra level of indenting aren't necessary, are they?  Why add them?